treeherder/push_health/push_health.py

```diff
@@ -69,25 +77,37 @@ def get_push_failures(push, option_map):
         test_key = '{}{}'.format(test_name, failure_line.job_guid)
         if test_name and test_key not in tests:
             job = failure_line.job_log.job
-            config = option_map[job.option_collection_hash]
+            config = clean_config(option_map[job.option_collection_hash])
+            platform = clean_platform(job.machine_platform.platform)
+            test_file = test_name.rsplit('/', 1)[-1]
+
+            # gather log lines.  If the test name isn't mentioned as
             errors = []
             for step in failure_line.job_log.job.text_log_step.all():
-                for error in step.errors.all():
-                    if len(errors) < 5:
-                        errors.append(error.line)
-            line = {
-                'testName': test_name,
-                'logLines': errors,
-                'jobName': job.job_type.name,
-                'jobId': job.id,
-                'jobClassificationId': job.failure_classification_id,
-                'platform': job.machine_platform.platform,
-                'suggestedClassification': 'New Failure',
-                'config': config,
-                'key': test_key,
-                'confidence': 0,
-            }
-            tests[test_key] = line
+                # only add log lines if the test_name is in the very first log_line
+                # and then only get up to 5 lines
+                log_lines = step.errors.all()
+                if test_file in log_lines.first().line:
+                    for idx, log_line in enumerate(log_lines[:5]):
+                        # don't add the log_line if it is in the ignore list
+                        if not any([ignored in log_line.line for ignored in ignored_log_lines]):
+                            errors.append(log_line.line)
```

Comment:
I think this is right, but the question I have is we are iterating through the failure_lines, then we check log_lines.  What is the differences?  Are they the same data set?

Reply:
I re-wrote this with using ``FailureLine``s instead of ``log_line``s.